how-to/workspace-platform-starter/client/src/modules/integrations/apps/integration.ts

```diff
@@ -364,52 +436,113 @@ export class AppProvider implements IntegrationModule<AppSettings> {
 	private async mapAppEntriesToSearchEntries(apps: PlatformApp[]): Promise<HomeSearchResult[]> {
 		const appResults: HomeSearchResult[] = [];
 		if (Array.isArray(apps)) {
-			const typeMapping = this._settings?.manifestTypeMapping;
-
-			for (const app of apps) {
-				const manifestType = app.manifestType;
-				if (isStringValue(manifestType)) {
-					const action = { name: "Launch View", hotkey: "enter" };
-					const entry: Partial<HomeSearchResult> = {
-						key: app.appId,
-						score: this._definition?.baseScore ?? AppProvider._DEFAULT_BASE_SCORE,
-						title: app.title,
-						data: { app, providerId: this._providerId }
-					};
-
-					if (!isEmpty(typeMapping)) {
-						const manifestTypeMapping = typeMapping[manifestType as ManifestTypeId];
-
-						if (!isEmpty(manifestTypeMapping)) {
-							if (isStringValue(manifestTypeMapping.entryLabel)) {
-								entry.label = manifestTypeMapping.entryLabel;
-							}
-							if (isStringValue(manifestTypeMapping.actionName)) {
-								action.name = manifestTypeMapping.actionName;
-							}
+			let favInfo: FavoriteInfo | undefined;
+			let favClient: FavoriteClient | undefined;
+			let savedFavorites: FavoriteEntry[] | undefined;
+			const colorScheme = (await this._integrationHelpers?.getCurrentColorSchemeMode()) ?? "dark";
+
+			if ((this._definition?.data?.favoritesEnabled ?? true) && this._integrationHelpers?.getFavoriteClient) {
+				favClient = await this._integrationHelpers.getFavoriteClient();
+				if (favClient) {
+					favInfo = favClient.getInfo();
+					if (favInfo.isEnabled) {
+						const isSupported = isEmpty(favInfo.enabledTypes) || favInfo.enabledTypes.includes("app");
+						if (isSupported) {
+							savedFavorites = await favClient.getSavedFavorites("app");
```

Comment:
should we have a const for the string "app" if we need to change it?

Reply:
or change FavoriteTypeNames to an enum